queue/driver_group_mgo.go

```diff
@@ -396,16 +396,16 @@ func (d *mgoGroupDriver) Next(ctx context.Context) amboy.Job {
 		},
 	}
 
+	timeLimits := bson.M{}
+	now := time.Now()
 	if d.opts.CheckWaitUntil {
-		qd = bson.M{
-			"$and": []bson.M{
-				qd,
-				{"$or": []bson.M{
-					{"time_info.wait_until": bson.M{"$lte": time.Now()}},
-					{"time_info.wait_until": bson.M{"$exists": false}}},
-				},
-			},
-		}
+		timeLimits["time_info.wait_until"] = bson.M{"$lte": now}
+	}
+	if d.opts.CheckDispatchBy {
+		timeLimits["time_info.dispatch_by"] = bson.M{"$gt": now}
```

Comment:
Does this need an index?

Reply:
perhaps, for now it's going to be off by default, and we can revisit as we turn on. I imagine it'll be ok to no use an index, but we can see.